### PR TITLE
Use trivy instead of owasp/dependency-check

### PR DESCRIPTION
**Pull request by jeffreyc-splunk**
Similar to the collector (https://github.com/signalfx/splunk-otel-collector/blob/main/.github/workflows/trivy-scans.yml), performs fs and image scans, and fails for high/critical vulnerabilities.

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -137,60 +137,40 @@ jobs:
...
-      - name: Uploading test result artifacts
-        uses: actions/upload-artifact@v3
+      - run: docker load -i ./dist/image.tar
+      - name: Run trivy image scan
+        uses: aquasecurity/trivy-action@0.9.1
         with:
-          name: dependency-check-result
-          path: reports
+          scan-type: 'image'
+          image-ref: 'quay.io/signalfx/signalfx-agent-dev:latest'
```
[the remark this answers is not in the document]
Comment (jeffreyc-splunk) [reply .github/workflows/build_and_test.yml:169]: Yes, that's the image name built from the upstream job: https://github.com/signalfx/signalfx-agent/blob/main/.github/workflows/build_and_test.yml#L39.

**Comment by rmfitzpatrick** [comment]
Is it worth creating an `output` artifact so we don't have to rely on job stdout?

**Comment by jeffreyc-splunk** [comment]
> Is it worth creating an [`output`](https://github.com/aquasecurity/trivy-action#inputs) artifact so we don't have to rely on job stdout?

I'll check it out to see what it'll look like.

**Comment by jeffreyc-splunk** [comment]
> > Is it worth creating an [`output`](https://github.com/aquasecurity/trivy-action#inputs) artifact so we don't have to rely on job stdout?
> 
> I'll check it out to see what it'll look like.

@rmfitzpatrick Updated to upload report to https://github.com/signalfx/signalfx-agent/security/code-scanning.

**Review by rmfitzpatrick — APPROVED** [review]

**Pull request merged**